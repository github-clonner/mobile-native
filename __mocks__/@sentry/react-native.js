export const captureMessage = jest.fn();
export default {};
