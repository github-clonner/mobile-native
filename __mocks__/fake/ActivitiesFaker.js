
function load(count, container) {

  let activities = [...Array(count)].map((_, i) => {
    const code = 'activityguid' + i;
    return {
      attachment_guid: false,
      blurb: false,
      container_guid:code,
      custom_data:false,
      custom_type:false,
      rowKey: 'something' + i,
      description:"Congratulations! ",
      edited:"",
      guid:code,
      mature:false,
      time_created: "1522036284",
      ownerObj:{
        guid: "824853017709780997",
        type: "user",
        subtype: false,
        time_created: "1522036284",
        // getAvatarSource: () => {
        //   return {
        //     source:'http://thisisaurl'
        //   }
        // }
      },
      shouldBeBlured: jest.fn(),
      message:"Message",
      title:'TITLE',
      owner_guid:"824853017709780997",
      parent_guid:"838106762591510528",
      perma_url:false,
      thumbnail_src:false,
      type:"activity",
      wire_totals: {
        tokens: 1000000000000000000
      },
      containerObj: container
      // _list: {
      //   viewed: {
      //     viewed: new Map([["1019155171608096768",true]]),
      //     addViewed: () => {
      //       return;
      //     }
      //   }
      // },
      // getThumbSource: () => {
      //   return {
      //     source:'http://thisisaurl'
      //   }
      // }
    }
  });

  return {
    'activities': activities,
    'load-next': 'aaaaaa',
    'load-previous': 'aaaaaa'
  }

}

/**
 * comment service data faker
 * @param {integer} count
 */
export function activitiesServiceFaker() {
  return { load: load}
}

