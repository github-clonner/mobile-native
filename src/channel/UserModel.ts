import { observable, action, runInAction } from 'mobx';

import { MINDS_CDN_URI, GOOGLE_PLAY_STORE } from '../config/Config';
import api from '../common/services/api.service';
import BaseModel from '../common/BaseModel';
import ChannelService from './ChannelService';
import sessionService from '../common/services/session.service';
import apiService from '../common/services/api.service';
import logService from '../common/services/log.service';

//@ts-nocheck
export const USER_MODE_OPEN = 0;
export const USER_MODE_MODERATED = 1;
export const USER_MODE_CLOSED = 2;

/**
 * User model
 */
export default class UserModel extends BaseModel {
  merchant;
  /**
   * Eth wallet
   */
  eth_wallet: string = '';
  wire_rewards;
  sums;
  btc_address?: string;
  icontime!: string;
  username!: string;
  name!: string;
  admin = false;
  plus: boolean = false;
  verified: boolean = false;
  founder: boolean = false;
  rewards: boolean = false;
  last_accepted_tos: number = 0;
  carousels?: Array<any>;

  /**
   * @var {boolean}
   */
  @observable blocked;

  /**
   * @var {number}
   */
  @observable subscribers_count;

  /**
   * @var {number}
   */
  @observable impressions;

  /**
   * @var {boolean}
   */
  @observable subscribed;

  /**
   * @var {boolean}
   */
  @observable mature_visibility = false;

  /**
   * @var {boolean}
   */
  @observable pending_subscribe = false;

  /**
   * @var {numeric}
   */
  @observable mode = 0;

  /**
   * @var {boolean}
   */
  @observable email_confirmed = false;

  /**
   * Confirm email
   * @param {Object} params
   */
  confirmEmail = async (params) => {
    // call any api endpoint with the param
    try {
      await apiService.get('api/v2/entities/', { urn: this.urn, ...params });
      this.setEmailConfirmed(true);
      return true;
    } catch (error) {
      return false;
    }
  };

  /**
   * Get the user icon time
   */
  getOwnerIcontime(): string {
    if (sessionService.getUser().guid === this.guid) {
      return sessionService.getUser().icontime;
    } else {
      return this.icontime;
    }
  }

  @action
  toggleMatureVisibility() {
    if (GOOGLE_PLAY_STORE) return;
    this.mature_visibility = !this.mature_visibility;
  }

  @action
  async toggleSubscription() {
    const value = !this.subscribed;
    this.subscribed = value;
    try {
      const metadata = this.getClientMetadata();
      await ChannelService.toggleSubscription(this.guid, value, metadata);
      UserModel.events.emit('toggleSubscription', this);
    } catch (err) {
      runInAction(() => {
        this.subscribed = !value;
      });
      throw err;
    }
  }

  @action
  async toggleBlock(value: boolean | null = null) {
    value = value === null ? !this.blocked : value;

    try {
      await ChannelService.toggleBlock(this.guid, value);
      this.blocked = value;
    } catch (err) {
      this.blocked = !value;
      logService.exception('[ChannelStore] toggleBlock', err);
    }
  }

  @action
  setMode(value) {
    this.mode = value;
  }

  @action
  setEmailConfirmed(value) {
    this.email_confirmed = value;
  }

  /**
   * Is admin
   */
  isAdmin() {
    return this.admin;
  }

  /**
   * current user is owner of the channel
   */
  isOwner = () => {
    return sessionService.getUser().guid === this.guid;
  };

  /**
   * Get banner source
   */
  getBannerSource() {
    if (this.carousels) {
      return {
        uri: this.carousels[0].src,
      };
    }
    return {
      uri: `${MINDS_CDN_URI}fs/v1/banners/${this.guid}/fat/${this.icontime}`,
      headers: api.buildHeaders(),
    };
  }

  /**
   * Get avatar source
   * @param {string} size
   */
  getAvatarSource(size = 'medium') {
    return {
      uri: `${MINDS_CDN_URI}icon/${
        this.guid
      }/${size}/${this.getOwnerIcontime()}`,
      headers: api.buildHeaders(),
    };
  }

  /**
   * Has banner
   * @returns {boolean}
   */
  hasBanner() {
    return !!this.carousels;
  }

  /**
   * Is closed
   * @returns {boolean}
   */
  isClosed() {
    return this.mode === USER_MODE_CLOSED;
  }

  /**
   * Is open
   * @returns {boolean}
   */
  isOpen() {
    return this.mode === USER_MODE_OPEN;
  }

  /**
   * Is moderated
   * @returns {boolean}
   */
  isModerated() {
    return this.mode === USER_MODE_MODERATED;
  }

  /**
   * Is subscribed
   * @returns {boolean}
   */
  isSubscribed() {
    return !!this.subscribed;
  }

  /**
   * Request subscribe
   */
  async subscribeRequest() {
    if (this.pending_subscribe || this.mode !== USER_MODE_CLOSED) {
      return;
    }
    try {
      this.pending_subscribe = true;
      await apiService.put(`api/v2/subscriptions/outgoing/${this.guid}`);
    } catch (err) {
      this.pending_subscribe = false;
      logService.exception(err);
    }
  }

  /**
   * Cancel subscribe request
   */
  async cancelSubscribeRequest() {
    if (!this.pending_subscribe || this.mode !== USER_MODE_CLOSED) {
      return;
    }
    try {
      this.pending_subscribe = false;
      await apiService.delete(`api/v2/subscriptions/outgoing/${this.guid}`);
    } catch (err) {
      this.pending_subscribe = true;
      logService.exception(err);
    }
  }
}
