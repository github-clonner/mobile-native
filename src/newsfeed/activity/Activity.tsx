import React, { Component } from 'react';

import { observer } from 'mobx-react';

import {
  Text,
  StyleSheet,
  TouchableOpacity,
  View,
  LayoutChangeEvent,
} from 'react-native';

import Icon from 'react-native-vector-icons/Ionicons';

import ExplicitText from '../../common/components/explicit/ExplicitText';
import OwnerBlock from './OwnerBlock';
import RemindOwnerBlock from './RemindOwnerBlock';
import Actions from './Actions';
import formatDate from '../../common/helpers/date';
import ActivityActionSheet from './ActivityActionSheet';
import ActivityEditor from './ActivityEditor';
import ActivityMetrics from './metrics/ActivityMetrics';
import MediaView from '../../common/components/MediaView';
import Translate from '../../common/components/Translate';
import ExplicitOverlay from '../../common/components/explicit/ExplicitOverlay';
import Lock from '../../wire/lock/Lock';
import { CommonStyle } from '../../styles/Common';
import Pinned from '../../common/components/Pinned';
import blockListService from '../../common/services/block-list.service';
import i18n from '../../common/services/i18n.service';
import ActivityModel from '../ActivityModel';
import BlockedChannel from '../../common/components/BlockedChannel';
import ThemedStyles from '../../styles/ThemedStyles';
import Colors from 'src/styles/Colors';

type PropsType = {
  entity: ActivityModel;
  navigation: any;
  hydrateOnNav?: boolean;
  isReminded?: boolean;
  autoHeight?: boolean;
  isLast?: boolean;
  hideTabs?: boolean;
  onLayout?: Function;
};

type StateType = {
  editing: boolean;
};

/**
 * Activity
 */
@observer
export default class Activity extends Component<PropsType, StateType> {
  /**
   * Translate reference
   */
  translate: Translate | null = null;

  /**
   * Remind reference
   */
  remind: Activity | null = null;

  /**
   * Remind reference
   */
  mediaView: MediaView | null = null;

  /**
   * initial state
   */
  state = {
    editing: false,
  } as StateType;
  /**
   * Nav to activity full screen
   */
  navToActivity = () => {
    if (!this.props.navigation) {
      return;
    }
    const navOpts = { entity: this.props.entity, hydrate: false };

    if (this.props.entity.remind_object || this.props.hydrateOnNav) {
      navOpts.hydrate = true;
    }

    this.props.navigation.push('Activity', navOpts);
  };

  /**
   * Navigate To channel
   */
  navToRemindChannel = () => {
    // only active if receive the navigation property
    if (this.props.navigation && this.props.entity.remind_object) {
      this.props.navigation.push('Channel', {
        guid: this.props.entity.remind_object.ownerObj.guid,
      });
    }
  };

  /**
   * On layout
   */
  onLayout = (e: LayoutChangeEvent) => {
    if (this.props.onLayout) {
      this.props.onLayout(e);
    }

    if (this.props.entity.listRef && this.props.entity._list) {
      const offsetToScrollTo =
        this.props.entity._list.scrollOffset + e.nativeEvent.layout.height;
      setTimeout(() => {
        this.props.entity.listRef?.scrollToOffset({
          offset: offsetToScrollTo,
          animated: true,
        });
        this.props.entity.listRef = undefined;
      }, 1000);
    }
  };

  /**
   * Render
   */
  render() {
    const entity = ActivityModel.checkOrCreate(this.props.entity);

    if (blockListService.blocked.has(entity.ownerObj.guid)) {
      return (
        <BlockedChannel entity={entity} navigation={this.props.navigation} />
      );
    }

    const hasText = !!entity.text;
    const lock =
      entity.paywall && entity.paywall === '1' ? (
        <Lock entity={entity} navigation={this.props.navigation} />
      ) : null;

    const message = this.state.editing ? (
      //Passing the store in newsfeed (could be channel also)
      <ActivityEditor entity={entity} toggleEdit={this.toggleEdit} />
    ) : (
      <View style={hasText ? styles.messageContainer : styles.emptyMessage}>
        {hasText ? (
          <ExplicitText
            entity={entity}
            navigation={this.props.navigation}
            style={styles.message}
          />
        ) : null}
        {hasText ? (
          <Translate
            ref={(r) => (this.translate = r)}
            entity={entity}
            style={styles.message}
          />
        ) : null}
      </View>
    );

    const show_overlay =
      entity.shouldBeBlured() &&
      !entity.is_parent_mature &&
      !(entity.shouldBeBlured() && entity.is_parent_mature);
    const overlay = show_overlay ? (
      <ExplicitOverlay entity={this.props.entity} />
    ) : null;

    const borderBottom = this.props.isReminded
      ? []
      : [ThemedStyles.style.borderBottomHair, ThemedStyles.style.borderPrimary];

    return (
      <View
        style={[
          styles.container,
          ...borderBottom,
          ThemedStyles.style.backgroundSecondary,
        ]}
        onLayout={this.onLayout}
        testID="ActivityView">
        <Pinned entity={this.props.entity} />
        {this.showOwner()}
        {lock}
        {message}
        <View>
          {this.showRemind()}

          <MediaView
            ref={(o) => {
              this.mediaView = o;
            }}
            entity={entity}
            navigation={this.props.navigation}
            style={styles.media}
            autoHeight={this.props.autoHeight}
          />
          {overlay}
        </View>
        {this.showActions()}
        {this.renderScheduledMessage()}
        {this.renderPendingMessage()}
        {this.renderActivitySpacer()}
        {this.renderActivityMetrics()}
      </View>
    );
  }

  /**
   * Render activity spacer
   */
  renderActivitySpacer = () => {
    return this.props.isLast ? <View style={styles.activitySpacer} /> : null;
  };

  /**
   * Render entity metrics
   */
  renderActivityMetrics = () => {
    return !this.props.hideTabs &&
      !this.props.entity.isScheduled() &&
      !this.props.entity.isPending() ? (
      <ActivityMetrics entity={this.props.entity} />
    ) : null;
  };

  /**
   * Show message if entity is scheduled
   */
  renderScheduledMessage = () => {
    return this.props.entity.isScheduled()
      ? this.renderYellowBanner(
          `${i18n.t('activity.scheduled')} ${formatDate(
            this.props.entity.time_created,
          )}.`,
        )
      : null;
  };

  /**
   * Show message if entity is awaiting moderation
   */
  renderPendingMessage = () => {
    return this.props.entity.isPending()
      ? this.renderYellowBanner(i18n.t('activity.pendingModeration'))
      : null;
  };

  /**
   * Render a banner with a message bellow the activity
   */
  renderYellowBanner = (message) => {
    return (
      <View style={[styles.yellowBanner, CommonStyle.padding]}>
        <Text style={[styles.yellowBannerText, CommonStyle.paddingLeft]}>
          {message}
        </Text>
      </View>
    );
  };

  /**
   * Pause video if exist
   */
  pauseVideo() {
    this.mediaView?.pauseVideo();
  }

  toggleEdit = (value) => {
    this.setState({ editing: value });
  };

  /**
   * Show translation
   */
  showTranslate = async () => {
    if (this.translate) {
      const lang = await this.translate.show();
      if (this.remind && lang) this.remind.showTranslate();
    } else {
      if (this.remind) this.remind.showTranslate();
    }
  };

  /**
   * Show Owner
   */
  showOwner() {
    if (!this.props.entity.remind_object) {
      const rightToolbar: React.ReactNode = (
        <View style={styles.rightToolbar}>
          <ActivityActionSheet
            toggleEdit={this.toggleEdit}
            entity={this.props.entity}
            navigation={this.props.navigation}
            onTranslate={this.showTranslate}
            testID={
              this.props.entity.text === 'e2eTest' ? 'ActivityMoreButton' : ''
            }
          />
        </View>
      );
      return (
        <OwnerBlock
          entity={this.props.entity}
          navigation={this.props.navigation}
          rightToolbar={this.props.hideTabs ? null : rightToolbar}>
          <TouchableOpacity
            onPress={() => this.navToActivity()}
            style={ThemedStyles.style.rowJustifyStart}>
            <Text
              style={[
                styles.timestamp,
                CommonStyle.paddingRight,
                ThemedStyles.style.colorSecondaryText,
              ]}>
              {formatDate(this.props.entity.time_created, 'friendly')}
            </Text>
            {this.props.entity.boosted && (
              <View style={styles.boostTagContainer}>
                <Icon
                  name="md-trending-up"
                  style={ThemedStyles.style.colorSecondaryText}
                />
                <Text
                  style={[
                    styles.boostTagLabel,
                    ThemedStyles.style.colorSecondaryText,
                  ]}>
                  {i18n.t('boosted').toUpperCase()}
                </Text>
              </View>
            )}
            {!!this.props.entity.edited && (
              <View style={styles.boostTagContainer}>
                <Text
                  style={[
                    styles.boostTagLabel,
                    ThemedStyles.style.colorSecondaryText,
                  ]}>
                  Â· {i18n.t('edited').toUpperCase()}
                </Text>
              </View>
            )}
          </TouchableOpacity>
        </OwnerBlock>
      );
    } else {
      return (
        <View>
          <RemindOwnerBlock
            entity={this.props.entity}
            navigation={this.props.navigation}
          />
          <View style={styles.rightToolbar}>
            {!this.props.hideTabs && (
              <ActivityActionSheet
                toggleEdit={this.toggleEdit}
                entity={this.props.entity}
                navigation={this.props.navigation}
                onTranslate={this.showTranslate}
                testID={
                  this.props.entity.text === 'e2eTest'
                    ? 'ActivityMoreButton'
                    : ''
                }
              />
            )}
          </View>
        </View>
      );
    }
  }

  /**
   * Show remind activity
   */
  showRemind() {
    const remind_object = this.props.entity.remind_object;

    if (remind_object) {
      if (blockListService.has(remind_object.owner_guid)) {
        return (
          <View
            style={[
              styles.blockedNoticeView,
              CommonStyle.margin2x,
              CommonStyle.borderRadius2x,
              CommonStyle.padding2x,
            ]}>
            <Text style={[CommonStyle.textCenter, styles.blockedNoticeDesc]}>
              {i18n.t('activity.remindBlocked')}
              <Text
                onPress={() => this.navToRemindChannel()}
                style={[CommonStyle.bold]}>
                {' '}
                @{remind_object.ownerObj.username}
              </Text>
            </Text>
          </View>
        );
      }

      if (this.props.entity.shouldBeBlured()) {
        remind_object.is_parent_mature = true;
      }

      return (
        <View style={styles.remind}>
          <Activity
            ref={(r) => (this.remind = r)}
            hideTabs={true}
            entity={remind_object}
            navigation={this.props.navigation}
            isReminded={true}
            hydrateOnNav={true}
          />
        </View>
      );
    }
  }

  /**
   * Show actions
   */
  showActions() {
    if (!this.props.hideTabs) {
      return (
        //@ts-ignore user store is injected
        <Actions
          entity={this.props.entity}
          navigation={this.props.navigation}
        />
      );
    }
  }
}

const styles = StyleSheet.create({
  container: {
    overflow: 'visible',
  },
  messageContainer: {
    padding: 10,
    paddingTop: 0,
  },
  message: {
    fontFamily: 'Roboto',
    fontSize: 15,
  },
  emptyMessage: {
    padding: 0,
  },
  media: {
    flex: 1,
  },
  timestamp: {
    fontSize: 12,
    color: '#888',
  },
  remind: {
    //  flex:1,
  },
  rightToolbar: {
    alignSelf: 'flex-end',
    position: 'absolute',
    right: 10,
    top: 20,
  },
  boostTagContainer: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  boostTagLabel: {
    fontWeight: '400',
    marginLeft: 2,
    fontSize: 10,
  },
  activitySpacer: {
    flex: 1,
    height: 70,
  },
  blockedNoticeView: {
    backgroundColor: '#eee',
  },
  blockedNoticeDesc: {
    opacity: 0.7,
  },
  yellowBannerText: {
    fontSize: 11,
    color: '#000',
  },
  yellowBanner: {
    backgroundColor: '#ffecb3',
  },
});
