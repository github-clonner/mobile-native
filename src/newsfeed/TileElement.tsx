//@ts-nocheck
import React, { PureComponent } from 'react';
import { TouchableOpacity, Platform } from 'react-native';
import FastImage from 'react-native-fast-image';

import { createImageProgress } from 'react-native-image-progress';
import ProgressCircle from 'react-native-progress/Circle';
import type ActivityModel from './ActivityModel';

const ProgressFastImage = createImageProgress(FastImage);

type PropsType = {
  entity: ActivityModel;
  size: number;
  navigation: any;
};

export default class TileElement extends PureComponent<PropsType> {
  /**
   * Navigate to view
   */
  _navToView = () => {
    if (this.props.navigation) {
      this.props.navigation.push('Activity', { entity: this.props.entity });
    }
  };

  render() {
    const style = { width: this.props.size, height: this.props.size };

    const source = this.props.entity.getThumbSource();

    const image =
      Platform.OS === 'android' ? (
        <ProgressFastImage
          indicator={ProgressCircle}
          source={source}
          style={style}
          threshold={150}
        />
      ) : (
        <FastImage source={source} style={style} />
      );

    return (
      <TouchableOpacity onPress={this._navToView} style={style}>
        {image}
      </TouchableOpacity>
    );
  }
}
