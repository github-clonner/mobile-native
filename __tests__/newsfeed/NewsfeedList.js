import 'react-native';
import React from 'react';
//import { Platform, CameraRoll, TouchableOpacity } from "react-native";
import { shallow } from 'enzyme';
import NewsfeedList from '../../src/newsfeed/NewsfeedList';
import { activitiesServiceFaker } from '../../__mocks__/fake/ActivitiesFaker';
import renderer from 'react-test-renderer';

import UserStore from '../../src/auth/UserStore';

import TileElement from '../../src/newsfeed/TileElement';
import Activity from '../../src/newsfeed/activity/Activity';
import UserModel from '../../src/channel/UserModel';
import ActivityModel from '../../src/newsfeed/ActivityModel';
jest.mock('../../src/channel/UserModel');
jest.mock('../../src/newsfeed/ActivityModel');
jest.mock('../../src/newsfeed/activity/Activity', () => 'Activity');
jest.mock('../../src/newsfeed/TileElement', () => 'TileElement');
jest.mock('../../src/auth/UserStore');

/**
 * Tests
 */
describe('newsfeed list screen component', () => {

  let store, entities, activityResponse, user;

  beforeEach(() => {
    activityResponse = activitiesServiceFaker().load(5);
    entities = activityResponse.activities;
    user = new UserStore();
    store = {
      refreshing: false,
      loadFeed: jest.fn()
    };
    store.list = {
      refreshing: false,
    };
    store.list.entities = entities;
    store.list.loaded = true;
    screen = shallow(
      <NewsfeedList.wrappedComponent newsfeed={store} user={user} />
    );

    screen.update();
  });

  it('should renders correctly', () => {
    screen.update();
    expect(screen).toMatchSnapshot();
  });


  it('should have n tileelements', async () => {

    activityResponse = activitiesServiceFaker().load(5);
    entities = activityResponse.activities;
    store.list = {
      refreshing: false,
    };
    store.list.entities = entities;
    store.list.loaded = true;
    store.isTiled = true;

    const testRenderer = renderer.create(
      <NewsfeedList.wrappedComponent  newsfeed={store} user={user}/>
    );
    let test = testRenderer.root;

    expect(test.findAllByType('Activity').length).toBe(0);

    expect(test.findAllByType('TileElement').length).toBe(5);

  });


  it('should have n activities', async () => {

    activityResponse = activitiesServiceFaker().load(5);
    entities = activityResponse.activities;
    store.list = {
      refreshing: false,
    };
    store.list.entities = entities;
    store.list.loaded = true;
    store.isTiled = false;

    const testRenderer = renderer.create(
      <NewsfeedList.wrappedComponent  newsfeed={store} user={user}/>
    );
    let test = testRenderer.root;

    expect(test.findAllByType('Activity').length).toBe(5);

    expect(test.findAllByType('TileElement').length).toBe(0);

  });

  
});