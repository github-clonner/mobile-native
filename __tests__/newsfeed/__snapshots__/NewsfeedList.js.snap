// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`newsfeed list screen component should renders correctly 1`] = `
<FlatList
  ListEmptyComponent={
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "height": 667,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#888",
              "fontFamily": "Roboto",
              "fontSize": 24,
              "fontWeight": "200",
              "marginBottom": 8,
              "marginTop": 8,
              "textAlign": "center",
            }
          }
        >
          Your newsfeed is empty
        </Text>
      </View>
    </View>
  }
  ListFooterComponent={null}
  data={
    Array [
      Object {
        "attachment_guid": false,
        "blurb": false,
        "containerObj": undefined,
        "container_guid": "activityguid0",
        "custom_data": false,
        "custom_type": false,
        "description": "Congratulations! ",
        "edited": "",
        "guid": "activityguid0",
        "mature": false,
        "message": "Message",
        "ownerObj": Object {
          "guid": "824853017709780997",
          "subtype": false,
          "time_created": "1522036284",
          "type": "user",
        },
        "owner_guid": "824853017709780997",
        "parent_guid": "838106762591510528",
        "perma_url": false,
        "rowKey": "something0",
        "shouldBeBlured": [MockFunction],
        "thumbnail_src": false,
        "time_created": "1522036284",
        "title": "TITLE",
        "type": "activity",
        "wire_totals": Object {
          "tokens": 1000000000000000000,
        },
      },
      Object {
        "attachment_guid": false,
        "blurb": false,
        "containerObj": undefined,
        "container_guid": "activityguid1",
        "custom_data": false,
        "custom_type": false,
        "description": "Congratulations! ",
        "edited": "",
        "guid": "activityguid1",
        "mature": false,
        "message": "Message",
        "ownerObj": Object {
          "guid": "824853017709780997",
          "subtype": false,
          "time_created": "1522036284",
          "type": "user",
        },
        "owner_guid": "824853017709780997",
        "parent_guid": "838106762591510528",
        "perma_url": false,
        "rowKey": "something1",
        "shouldBeBlured": [MockFunction],
        "thumbnail_src": false,
        "time_created": "1522036284",
        "title": "TITLE",
        "type": "activity",
        "wire_totals": Object {
          "tokens": 1000000000000000000,
        },
      },
      Object {
        "attachment_guid": false,
        "blurb": false,
        "containerObj": undefined,
        "container_guid": "activityguid2",
        "custom_data": false,
        "custom_type": false,
        "description": "Congratulations! ",
        "edited": "",
        "guid": "activityguid2",
        "mature": false,
        "message": "Message",
        "ownerObj": Object {
          "guid": "824853017709780997",
          "subtype": false,
          "time_created": "1522036284",
          "type": "user",
        },
        "owner_guid": "824853017709780997",
        "parent_guid": "838106762591510528",
        "perma_url": false,
        "rowKey": "something2",
        "shouldBeBlured": [MockFunction],
        "thumbnail_src": false,
        "time_created": "1522036284",
        "title": "TITLE",
        "type": "activity",
        "wire_totals": Object {
          "tokens": 1000000000000000000,
        },
      },
      Object {
        "attachment_guid": false,
        "blurb": false,
        "containerObj": undefined,
        "container_guid": "activityguid3",
        "custom_data": false,
        "custom_type": false,
        "description": "Congratulations! ",
        "edited": "",
        "guid": "activityguid3",
        "mature": false,
        "message": "Message",
        "ownerObj": Object {
          "guid": "824853017709780997",
          "subtype": false,
          "time_created": "1522036284",
          "type": "user",
        },
        "owner_guid": "824853017709780997",
        "parent_guid": "838106762591510528",
        "perma_url": false,
        "rowKey": "something3",
        "shouldBeBlured": [MockFunction],
        "thumbnail_src": false,
        "time_created": "1522036284",
        "title": "TITLE",
        "type": "activity",
        "wire_totals": Object {
          "tokens": 1000000000000000000,
        },
      },
      Object {
        "attachment_guid": false,
        "blurb": false,
        "containerObj": undefined,
        "container_guid": "activityguid4",
        "custom_data": false,
        "custom_type": false,
        "description": "Congratulations! ",
        "edited": "",
        "guid": "activityguid4",
        "mature": false,
        "message": "Message",
        "ownerObj": Object {
          "guid": "824853017709780997",
          "subtype": false,
          "time_created": "1522036284",
          "type": "user",
        },
        "owner_guid": "824853017709780997",
        "parent_guid": "838106762591510528",
        "perma_url": false,
        "rowKey": "something4",
        "shouldBeBlured": [MockFunction],
        "thumbnail_src": false,
        "time_created": "1522036284",
        "title": "TITLE",
        "type": "activity",
        "wire_totals": Object {
          "tokens": 1000000000000000000,
        },
      },
    ]
  }
  disableVirtualization={false}
  horizontal={false}
  initialNumToRender={6}
  key="f"
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReached={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onRefresh={[Function]}
  onViewableItemsChanged={[Function]}
  refreshing={false}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  style={
    Array [
      undefined,
      undefined,
    ]
  }
  updateCellsBatchingPeriod={50}
  user={
    Object {
      "clearUser": [MockFunction],
      "hasAttachment": false,
      "isAdmin": [MockFunction],
      "load": [MockFunction],
      "me": Object {},
      "reset": [MockFunction],
      "setRewards": [MockFunction],
      "setUser": [MockFunction],
    }
  }
  viewabilityConfig={
    Object {
      "minimumViewTime": 300,
      "viewAreaCoveragePercentThreshold": 50,
    }
  }
  windowSize={11}
/>
`;
